src/main/java/sjchat/restapi/controllers/ChatController.java

```diff
@@ -0,0 +1,253 @@
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import sjchat.messages.ChatDataRequest;
+import sjchat.messages.ChatListRequest;
+import sjchat.messages.ChatListResponse;
+import sjchat.messages.ChatResponse;
+import sjchat.messages.MessageListRequest;
+import sjchat.messages.MessageListResponse;
+import sjchat.messages.MessageResponse;
+import sjchat.messages.MessageServiceGrpc;
+import sjchat.messages.NewMessageRequest;
+import sjchat.restapi.entities.Chat;
+import sjchat.restapi.entities.ChatRequest;
+import sjchat.restapi.entities.Message;
+import sjchat.restapi.entities.User;
+import sjchat.users.UserDataRequest;
+import sjchat.users.UserRequest;
+import sjchat.users.UserResponse;
+import sjchat.users.UserServiceGrpc;
+
+@RestController
+public class ChatController {
+  private Channel messageServiceChannel;
+  private Channel userServiceChannel;
+
+  public ChatController() {
+    messageServiceChannel = buildMessageServiceChannel();
+    userServiceChannel = buildUserServiceChannel();
+  }
+
+  private static Channel buildMessageServiceChannel() {
+    return ManagedChannelBuilder.forAddress("localhost", 50052).usePlaintext(true).build(); //TODO: Put port in config file
+  }
+
+  private static Channel buildUserServiceChannel() {
+    return ManagedChannelBuilder.forAddress("localhost", 50051).usePlaintext(true).build(); //TODO: Put port in config file
+  }
+
+  private static Chat buildChatFromResponse(sjchat.messages.Chat responseChat) {
+    Chat chat = new Chat();
+    chat.setId(responseChat.getId());
+    chat.setTitle(responseChat.getTitle());
+
+    List<sjchat.users.User> chatUsers = responseChat.getUsersList();
+    for (sjchat.users.User chatUser : chatUsers) {
+      User user = buildUserFromResponse(chatUser);
+      chat.addUser(user);
+    }
+
+    return chat;
+  }
+
+  private static Message buildMessageFromResponse(sjchat.messages.Message responseMessage) {
+    Message message = new Message();
+    message.setId(responseMessage.getId());
+    message.setMessage(responseMessage.getMessage());
+    message.setUser(responseMessage.getUser());
+
+    return message;
+  }
+
+  private static User buildUserFromResponse(sjchat.users.User responseUser) {
+    User user = new User();
+    user.setId(responseUser.getId());
+    user.setUsername(responseUser.getUsername());
+
+    return user;
+  }
+
+  @RequestMapping(
+          value = "/chat",
+          method = RequestMethod.GET,
+          produces = "application/json")
+  @ResponseBody
+  public ResponseEntity<List<Chat>> getChatList() {
+    final MessageServiceGrpc.MessageServiceBlockingStub blockingStub = MessageServiceGrpc.newBlockingStub(messageServiceChannel);
+
+    ArrayList<Chat> chatList = new ArrayList<>();
```

Comment:
Also, as a general comment: the Google core libs for Java https://github.com/google/guava includes some neat immutable data structures (yay, immutability!), which we should consider using.